connector-pagerduty/src/main/java/com/redhat/cloud/notifications/connector/pagerduty/PagerDutyCloudEventDataExtractor.java

```diff
@@ -0,0 +1,90 @@
+package com.redhat.cloud.notifications.connector.pagerduty;
+
+import com.redhat.cloud.notifications.connector.CloudEventDataExtractor;
+import com.redhat.cloud.notifications.connector.authentication.AuthenticationDataExtractor;
+import io.vertx.core.json.JsonObject;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.apache.camel.Exchange;
+import org.apache.camel.http.common.HttpMethods;
+import org.apache.commons.validator.routines.UrlValidator;
+import org.apache.http.ProtocolException;
+
+import java.util.MissingResourceException;
+
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.TARGET_URL;
+import static com.redhat.cloud.notifications.connector.pagerduty.ExchangeProperty.ACCOUNT_ID;
+import static com.redhat.cloud.notifications.connector.pagerduty.ExchangeProperty.TARGET_URL_NO_SCHEME;
+import static org.apache.commons.validator.routines.UrlValidator.ALLOW_LOCAL_URLS;
+
+@ApplicationScoped
+public class PagerDutyCloudEventDataExtractor extends CloudEventDataExtractor {
+
+    public static final String NOTIF_METADATA = "notif-metadata";
+    public static final String AUTHENTICATION = "authentication";
+    public static final String METHOD = "method";
+    public static final String TRUST_ALL = "trustAll";
+    public static final String URL = "url";
+
+    public static final String PAYLOAD = "payload";
+    public static final String CUSTOM_DETAILS = "custom_details";
+    public static final String EVENT_ACTION = "event_action";
+
+    private static final UrlValidator HTTPS_URL_VALIDATOR = new UrlValidator(new String[]{"https"}, ALLOW_LOCAL_URLS);
+    private static final UrlValidator HTTP_URL_VALIDATOR = new UrlValidator(new String[]{"http"}, ALLOW_LOCAL_URLS);
```

Comment:
For the rest of the connectors we make sure that the target URLs don't point to an internal IP, so I think we can remove the `ALLOW_LOCAL_URLS` setting from here.

Reply:
so fun story, removing this actually breaks 6 of the test cases in `PagerDutyConnectorRoutesTest`, because the mock server URL is `localhost` 🙃. Replacing the host with an IP address also fails because TLS certs can't be issued to IP addresses...

Adding an additional `Exchange` param to check if a test case is being run would work, but that feels icky so I was wondering what you think.

Reply:
Yeah, now that I think about it I guess this should be validated whenever the user creates a new Pager Duty integration through the UI. That's what we do for the webhooks, for example.

Reply:
I'll add it back in for now. I don't think users will be setting the URL themselves anyways since the same endpoint is used for everyone.